src/WinRT.Runtime/Projections/ICustomPropertyProvider.net5.cs

```diff
@@ -289,25 +385,34 @@ static unsafe ManagedCustomPropertyProviderVftbl()
 
 
         [UnmanagedCallersOnly]
-
         private static unsafe int Do_Abi_GetCustomProperty_0(IntPtr thisPtr, IntPtr name, IntPtr* result)
         {
             global::Microsoft.UI.Xaml.Data.ICustomProperty __result = default;
             try
             {
                 string _name = MarshalString.FromAbi(name);
-                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
-                PropertyInfo propertyInfo = target.GetType().GetProperty(
-                     _name,
-                     BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public);
-
-                if (propertyInfo is object)
-                {
-                    __result = new ManagedCustomProperty(propertyInfo);
+                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
+
+                if (target is global::Microsoft.UI.Xaml.Data.IBindableCustomPropertyImplementation bindableCustomPropertyImplementation)
+                {
+                    __result = bindableCustomPropertyImplementation.GetProperty(_name);
+                    *result = MarshalInterface<global::Microsoft.UI.Xaml.Data.ICustomProperty>.FromManaged(__result);
+                    return 0;
+                }
+
+                if (RuntimeFeature.IsDynamicCodeCompiled)

```

Comment:
Should we not immediately throw `NotSupportedException` here if on AOT and the target doesn't implement the interface, and include an error message to explain the attribute is missing on that type for this to work?

Reply:
That is an interesting proposal, will try it out and test it with WinUI gallery to see if this doesn't cause any unexpected issues.

Reply:
The reason why I'm suggesting that is that otherwise you'd have no way of removing the trim warnings and also make the behavior testable and reliable. Essentially, without it you'd be back in undefined behavior land, as you'd either still have those trim warnings during publish, or you'd have the code potentially crash at runtime due to no metadata.

Reply:
Addressed, will see how it works out with WinUI Gallery.

Reply:
Things seem to be good with WinUI gallery testing, no unexpected issues.  It was actually useful in guiding to find the types to address.